### PR TITLE
Migrate from webpack to Vite

### PR DESCRIPTION
**Pull request by reillyeon**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -278,7 +266,9 @@ async function connectToPort(): Promise<void> {
     connectButton.disabled = false;
   } catch (e) {
     console.error(e);
-    term.writeln(`<ERROR: ${e.message}>`);
+    if (e instanceof Error) {
```
Comment (tomayac) [inline src/index.ts:269]: Gotta love TypeScript…

File: vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            purpose: "any maskable"
+          },
+          {
+            src: "images/icons-512.png",
+            type: "image/png",
+            sizes: "512x512",
+            purpose: "any maskable"
+          }
+        ],
+        "start_url": "./?source=pwa",
```
Comment (tomayac) [inline vite.config.ts:31]: You should also add an `"id"`.

**Review by tomayac — APPROVED** [review]
LGTM, with some suggestions.

File: src/index.ts (hunk 1 above)
Comment (reillyeon) [reply src/index.ts:269]: This one is on JavaScript for not requiring things that are thrown to be errors.

File: vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import eslint from 'vite-plugin-eslint'
+import { VitePWA } from 'vite-plugin-pwa'
+
+export default {
+  plugins: [
+    eslint(),
+    VitePWA({
```
[the remark this answers is not in the document]
Comment (reillyeon) [reply vite.config.ts:7]: I'm leaving out enabling the service worker in development for now because it doesn't seem to play well with vite-eslint-plugin.

**Pull request merged**